examples/utils_data.jl

```diff
@@ -428,46 +428,48 @@ function get_data_correlated_ou2d_timeseries(batchsize;
                                  
     @assert xor(read, save)
 
+    # TODO: this is an extremly memory inefficient way of creating the dataloaders
     rawtrain = CliMADatasets.CorrelatedOU2D(:train; f = f, resolution=resolution, Tx=FT);
     samp_per_τ = Int(round(rawtrain.metadata["τ"]/ rawtrain.metadata["dt_save"]))
     rawtrain = rawtrain.features
     rawtest = CliMADatasets.CorrelatedOU2D(:test; f = f, resolution=resolution, Tx=FT)[:]
 
     # Create train and test datasets
+    stride = Int(samp_per_τ)
+    @assert stride > ntime
 
-    @assert pairs_per_τ <= samp_per_τ/2
-    stride = Int(floor(samp_per_τ/pairs_per_τ))
-    
     nsamp_train = size(rawtrain)[end]
-    npairs_train = length(Array(1:stride:(nsamp_train-1)))
-    xtrain  = zeros(FT, (resolution, resolution, 2, npairs_train))
-    xtrain[:,:,1,:] .= rawtrain[:,:,1:stride:(nsamp_train-1)]
-    xtrain[:,:,2,:] .= rawtrain[:,:,2:stride:nsamp_train]
+    nseries_train = div(nsamp_train, stride) 
+    xtrain  = zeros(FT, (resolution, resolution, ntime, 1, nseries_train))
+    for i in 1:ntime
+        xtrain[:,:,i,1,:] .= rawtrain[:,:,i:stride:(nsamp_train-rem(nsamp_train, stride))]
+    end
 
     nsamp_test = size(rawtest)[end]
-    npairs_test = length(Array(1:stride:(nsamp_test-1)))
-    xtest  = zeros(FT, (resolution, resolution, 2, npairs_test))
-    xtest[:,:,1,:] .= rawtest[:,:,1:stride:(nsamp_test-1)]
-    xtest[:,:,2,:] .= rawtest[:,:,2:stride:nsamp_test]
+    nseries_test = div(nsamp_test, stride)
+    xtest  = zeros(FT, (resolution, resolution, ntime, 1, nseries_test))
+    for i in 1:ntime
+        xtest[:,:,i,1,:] .= rawtest[:,:,i:stride:(nsamp_test-rem(nsamp_test, stride))]
+    end
 
     # Scaling
     if save
         if standard_scaling
-            maxtrain = maximum(xtrain, dims=(1, 2, 4))
-            mintrain = minimum(xtrain, dims=(1, 2, 4))
+            maxtrain = maximum(xtrain, dims=(1, 2, 3, 5))
+            mintrain = minimum(xtrain, dims=(1, 2, 3, 5))
             Δ = maxtrain .- mintrain
             # To prevent dividing by zero
             Δ[Δ .== 0] .= FT(1)
             scaling = StandardScaling{FT}(mintrain, Δ)
         else
             #scale means and spatial variations separately
             x̄ = mean(xtrain, dims=(1, 2))
```

Comment:
hmmmmm im wondering now if we want this to be the mean over dims 1,2,3? isn't that the equivalent of the "Spatial mean" in 3d?

Reply:
i.e. couldnt we get a color shift in time?

Reply:
true!

Reply:
nice catch :)